Update issue templates (#27570)
WordPress/gutenberg: .github/ISSUE_TEMPLATE/Bug_report.md

---
name: Bug report
about: Report a bug with the WordPress block editor or Gutenberg plugin

---

<!--

> Please fill out ALL sections. Bug reports with missing information will be

Author: In reality, we usually just ask the reporter for more information 🙂

Reviewer: Nitpick - Should it say all required sections?

Could also make the 'ALL' even louder:
```
**ALL**
```

or 
```
<span style="font-size:99999999px">ALL</span>
```

Author: I think capitalising is enough. This is only shown when you're in the "Write" tab so any markdown and/or HTML won't be rendered.

```suggestion
Please fill out ALL required sections. Bug reports with missing information will be
```

closed.

Before submitting a bug report:

- Check if the bug has already been fixed by updating WordPress and/or Gutenberg.
- Check if the bug is caused by a plugin by deactivating all plugins except Gutenberg.

> - Check if the bug is caused by a theme by activating a default theme e.g. Twenty Twenty.

Author: This preamble is intended to encourage the reporter to do a little upfront investigation. This makes bug reports easier to reproduce, triage, and eventually fix.

> - Check if the bug is caused by a theme by activating a default theme e.g. Twenty Twenty.

Reviewer 2: Should we also add "Check if the bug is already reported in another issues"?

Reviewer 3: I think it would make sense to add this in too!

If this is a security issue, please report it in HackerOne instead:
https://hackerone.com/wordpress
-->

> **Description**

Reviewer: I need to follow-up by changing these to proper headings, it's been bugging me for ages that they're not.

Author: It would be more accessible too. Want to push up a commit which does that? 🙂

Reviewer: Done

<!-- Please write a brief description of the bug. -->

**Step-by-step reproduction instructions**
<!--
Please list the steps needed to reproduce the bug. For example:
1. Go to '...'
2. Click on '...'
3. Scroll down to '...'
-->

**Expected behaviour**
<!-- Please describe what you expected to happen. -->

**Actual behaviour**
<!-- Please describe what actually happened. -->

**Screenshots or screen recording (optional)**
<!--
If possible, please upload a screenshot or screen recording which demonstrates
the bug. You can use LIEcap to create a GIF screen recording:

> https://www.cockos.com/licecap/

Author: Uploading a GIF screen recording helps a lot with triaging, so we should try to encourage it. LICEcap is free and open source software.

> -->

Author: Uploading a GIF screen recording helps a lot with triaging, so we should try to encourage it. LICEcap is free and open source software.

Reviewer 2: I use [Kap](https://getkap.co/), which is also free and open source, and built with web technology 😆

Reviewer 4: (Rephrasing above sentence.)
Example GIF screen recording software such as https://www.cockos.com/licecap/ or https://getkap.co/ or any other software can be used.

Author: Does Kap work on Windows?

Reviewer 4: From looking at kap web site it looks like Mac only software.
I went ahead and asked them if they have plans to add a version for Windows PC's: https://github.com/wulkano/Kap/issues/969

Reviewer 4: Here is another program I have heard a mention of earlier. Which works on Mac and Windows.
https://monosnap.com/welcome

**Plugin code (optional)**
<!--
If this bug is to do with a developer API, please share a code snippet or plugin
code that demonstrates the issue. You can use GitHub Gist to share multiple code
files:
https://gist.github.com

> -->

Author: It's very tough to triage bug reports to do with a developer API when no code sample is provided, so this is intended to encourage that.

Reviewer: I'm not sure about the phrase 'Plugin code', I think a developer would understand something like 'Code snippet' without that.

I think we should also add something here to make sure they don't just provide pseudocode, I've seen a few of those and they're still hard to interpret. Ideally it's code we can paste into the console or file in our local env that we can use to test the bug.

I considered the phrase 'please share a working code snippet', but that might be confusing if it's code that triggers a bug.

Author: Yeah, fair enough. What do you suggest we change this to? (Could you write a ` ```suggestion` for me to commit? 🙂 )

Ideally I'd love for us to have a Codepen template or something which can be used to demonstrate this kind of stuff. Not sure how feasible it is that one Codepen template could cover all APIs, though.

> ## Plugin code (optional)
> <!--
> If this bug is to do with a developer API, please share a code snippet or plugin
> code that demonstrates the issue. You can use GitHub Gist to share multiple code
> files:
> https://gist.github.com
> -->

Reviewer: ```suggestion
## Code snippet (optional)
<!--
If this bug is to related to a developer API, please share a code snippet that 
demonstrates the issue.

For small snippets paste it directly here, or you can use GitHub Gist to share
multiple code files: https://gist.github.com

Please ensure the shared code can be used by a developer to reproduce the 
issue—ideally it can be copied into a local development environment or executed
in a browser console to help debug the issue.
-->
```

Reviewer: @noisysocks How does that sound?

Author: Very good!

**WordPress information**
- WordPress version: <!-- e.g. "5.6.0" -->
- Gutenberg version: <!-- e.g. "9.4.0" or "Not installed" -->
- Are all plugins except Gutenberg deactivated? <!-- "Yes" or "No" -->

> - Are you using a default theme (e.g. Twenty Twenty)? <!-- "Yes" or "No" -->

Author: Similar to the preamble. This little bit of upfront investigation helps with reproducing, triaging, and fixing.

> - Are you using a default theme (e.g. Twenty Twenty)? <!-- "Yes" or "No" -->

Reviewer 5: ```suggestion
- Are you using a default theme (e.g. Twenty Twenty One)? <!-- "Yes" or "No" -->
```
😎

Author: ```suggestion
- Are you using a default theme (e.g. Twenty Twenty-One)? <!-- "Yes" or "No" -->
```

> - Are you using a default theme (e.g. Twenty Twenty-One)? <!-- "Yes" or "No" -->

Reviewer 2: Probably not worth putting it here, but maybe adding an instruction of how to access these properties would help in some ways? Plugins and themes are often straight-forward to find, but WordPress version is often trickier for new-comers. I'd suggest to just add a link after it. WDYT?

Reviewer 3: Could say, "You can find this information in your wp-admin dashboard under Tools > Site Health > Info".

Reviewer 4: Usually we also find the WordPress version information at the bottom of the screen. But adding a link to Site Health feels in a sense more proper.

## Device information
- Device: <!-- e.g. "Desktop" or "iPhone 11" -->
- Operating system: <!-- e.g. "Windows 10" or "iOS 14" -->

> - Browser: <!-- e.g. "Chrome 86.0" or "Mobile Safari" -->

Reviewer 5: It might be a WordPress mobile app as well.  @hypest and @Tug, do you report bugs in this repository? I guess you do 😃

Reviewer 6: Yeap, we do report native mobile issues in this repo too 👍

Maybe we can change this line to something like `Client: <!-- e.g. "Chrome 86.0" or "Mobile Safari" or "WordPress Android" or "WordPress iOS" -->`?

Author: I noticed we have a seperate bug report template for the mobile apps. Could we consolidate into one?

Reviewer 6: > Could we consolidate into one?

Yes, that makes sense, thanks Robert! Consolidating in this very PR could do it I think.

While at it, may I suggest introducing a "Step 0. On the <client>" on the steps to reproduce, to state which client the steps are about? That might help with staying in the right context while reading the steps. WDYT?

Author: I decided not to consolidate the two templates because having them separated means that GitHub can automatically add the `Mobile App Android/iOS` label which I assume is desirable.

I did update the mobile bug report template based on our changes to the web bug report template, through.

Reviewer 6: Good point about the label adding. Keeping them separate can still work and we can always revise if need arises. Thanks!
